File: app/src/main/java/ru/deltadelete/lab10/entities/CountryWithTowns.java
package ru.deltadelete.lab10.entities;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class CountryWithTowns {
    @Embedded public Country country;
    @Relation(parentColumn = "country_id", entityColumn = "country_id")
    public List<Town> towns;
}

File: app/src/main/java/ru/deltadelete/lab10/entities/CountryWithTowns.kt
package ru.deltadelete.lab10.entities

import androidx.room.Embedded
import androidx.room.Relation

public data class CountryWithTowns (
    @Embedded
    var country: Country? = null,

    @Relation(parentColumn = "country_id", entityColumn = "country_id")
    var towns: List<Town>? = null,
)
